nanoFramework.System.Net/IPAddress.cs

```diff
@@ -24,12 +24,38 @@ public class IPAddress
         /// <summary>
         /// Provides an IP address that indicates that the server must listen for client activity on all network interfaces. This field is read-only.
         /// </summary>
-        public static readonly IPAddress Any = new(0x0000000000000000);
+        /// <remarks>
+        /// The <see cref="Socket.Bind"/> method uses the <see cref="Any"/> field to indicate that a <see cref="Socket"/> instance must listen for client activity on all network interfaces.
+        /// 
+        /// The <see cref="Any"/> field is equivalent to 0.0.0.0 in dotted-quad notation.
+        /// </remarks>
+        public static readonly IPAddress Any = new(new byte[] { 0, 0, 0, 0 });
```

Comment:
Why this change? The byte[] constructor is soooooo much more inefficient. 🤔

Reply:
I thought we had a discussion earlier regarding avoiding magic numbers and came to the conclusion that the byte[] constructor or even the "Parse" method were clearer. 

Since these are all static readonly I'm not sure the efficiency is much concern.

Reply:
Forgive me but I can't recall that conversation... 😅
Anyway, both approaches are declaring the IP address to be used in the code, so that these are made available and no magic numbers are used in the code. 
At the declaration level, these numbers will always have to exist right? Either as byte array elements or integers...

BTW, most code, as usually happens was taken/inspired from the source code of the full .NET:
https://referencesource.microsoft.com/#System/net/System/Net/IPAddress.cs,18

Reply:
.NET Core went the other way any used the byte array constructors: https://github.com/dotnet/runtime/blob/342951dec5dc326a4b0ba0d0949b7a4155154e90/src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs#L20

I'm fine either way just let me know what we prefer.

Reply:
Interesting... well, nano code base is comming from 4.x so no surprise this was using that approach.
No need to put more work on this. OK with byte arrays. I was just checking why the change.